### PR TITLE
Features dev

### PR DESCRIPTION
**Pull request by calebbratton**
fixed merge conflicts w/ development

### CODE REVIEW

File: client/src/components/Flights/FlightController.jsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+  const getOneWayTrip = () => {
+    axios.get('api/flights/oneWay', {
+      params: {
+        originLocationCode: flightInfo.originLocationCode,
+        destinationLocationCode: flightInfo.destinationLocationCode,
+        departureDate: flightInfo.departureDate,
+        adults: flightInfo.adults,
+        currencyCode: 'USD',
+        maxPrice: 1000,
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/FlightController.jsx:45]: Should change in future to user's budget

File: client/src/components/Flights/FlightController.jsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      setNonStop(true);
+    }
+  }
+
+  function roundTripSwitch() {
+    if (roundTrip === true) {
+      setRoundTrip(false);
+    } else if (roundTrip === false) {
+      setRoundTrip(true);
+    }
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/FlightController.jsx:83]: could do `setRoundTrip(!roundTrip)`

File: client/src/components/Flights/FlightTile.jsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  const classes = useStyles();
+  const [fav, setFav] = useState(false);
+  const { setFavorites, favorites } = useContext(AppContext);
+
+  const displaySegments = (itinerary, arriveDepart) => {
+    itinerary.segments.map((segment) => {
+      const arrivalTime = segment[arriveDepart].at;
+      const date = new Date(arrivalTime);
+      return (
+        console.log('segment', segment[arriveDepart].iataCode, date)
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/FlightTile.jsx:43]: TODO

File: client/src/components/Flights/FlightTile.jsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      );
+    });
+  };
+
+  const saveFlight = (favFlight) => {
+    axios.put('/user/flights', favFlight)
+      .then((response) => {
+        console.log(response);
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/FlightTile.jsx:51]: no need for this console.log

File: client/src/components/Flights/FlightTile.jsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                {flight.itineraries.length < 2 ? (
+                  <Typography variant="body2" gutterBottom>
+                    Departure:
+                    {' '}
+                    {displaySegments(flight.itineraries[0], 'departure')}
+                    <br />
+                    Arrival:
+                    {' '}
+                    {displaySegments(flight.itineraries[0], 'arrival')}
+                  </Typography>
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/FlightTile.jsx:82]: could change to a map for these items

File: client/src/components/Flights/FlightTile.jsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            <Grid item xs container direction="column" spacing={2}>
+              <Grid item xs>
+                <Typography gutterBottom variant="subtitle1">
+                  Standard license
+                </Typography>
+                {flight.itineraries.length < 2 ? (
+                  <Typography variant="body2" gutterBottom>
+                    Departure:
+                    {' '}
+                    {displaySegments(flight.itineraries[0], 'departure')}
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/FlightTile.jsx:77]: String templating!!!

File: client/src/components/Flights/FlightTile.jsx [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+                <Fab
+                  aria-label="like"
+                  size="small"
+                  onClick={(e) => {
+                    e.preventDefault();
+                    setFav(true);
+                    saveFlight(flight);
+                    setFavorites([...favorites, flight]);
+                    console.log('FLIGHT', flight);
+                  }}
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/FlightTile.jsx:132]: could extract to new function

File: client/src/components/Flights/Flights.jsx [hunk 8]
```diff
@@ -1,32 +1,28 @@
...
 
 export default function Flights() {
+  const { flights, setFlights } = useContext(AppContext);
+  const [nonStop, setNonStop] = useState(false);
+  const [roundTrip, setRoundTrip] = useState(false);
   const classes = useStyles();
+
+  useEffect(() => {
+    console.log(flights[0]);
+  }, [flights]);
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/Flights.jsx:16]: seems unnecessary

File: client/src/components/Flights/Flights.jsx [hunk 9]
```diff
@@ -1,32 +1,28 @@
...
+import { CircularProgress } from '@material-ui/core';
+import useStyles from '../TabStyle.jsx';
+import { AppContext } from '../../helpers/context';
+import FlightController from './FlightController.jsx';
+import FlightTile from './FlightTile.jsx';
 
 export default function Flights() {
+  const { flights, setFlights } = useContext(AppContext);
+  const [nonStop, setNonStop] = useState(false);
+  const [roundTrip, setRoundTrip] = useState(false);
```
Comment (Pix3lAssassin) [inline client/src/components/Flights/Flights.jsx:11]: could be put in flights controller without consequence

File: client/src/components/Hotels/HotelController.jsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
```
Comment (Pix3lAssassin) [inline client/src/components/Hotels/HotelController.jsx:1]: useState is unused

File: client/src/components/Hotels/HotelDatePicker.jsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+    if (month.length < 2) {
+      month = '0' + month;
+    }
+    if (day.length < 2) {
+      day = '0' + day;
+    }
+    return [year, month, day].join('-');
+  }
+  const onDateChange = (startDate, endDate) => {
```
Comment (Pix3lAssassin) [inline client/src/components/Hotels/HotelDatePicker.jsx:20]: pretty sure this is duplicate code could be extracted to a new file

File: client/src/components/Hotels/HotelTile.jsx [hunk 12]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import ButtonBase from '@material-ui/core/ButtonBase';
+
```
Comment (Pix3lAssassin) [inline client/src/components/Hotels/HotelTile.jsx:7]: could use destructuring for this - P3

File: client/src/components/Hotels/HotelTile.jsx [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+              <Grid item xs>
+                <Typography gutterBottom variant="subtitle1">
+                  {hotel.hotel.name}
+                </Typography>
+                <Typography variant="body2" gutterBottom>
+                  {hotel.hotel.description ? hotel.hotel.description.text
+                    : <span>No Description</span>}
+                </Typography>
+                <Typography variant="body2" color="textSecondary">
+                  ID: 1030114
```
Comment (Pix3lAssassin) [inline client/src/components/Hotels/HotelTile.jsx:53]: TODO

File: client/src/components/Hotels/HotelTile.jsx [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+            </ButtonBase>
+          </Grid>
+          <Grid item xs={12} sm container>
+            <Grid item xs container direction="column" spacing={2}>
+              <Grid item xs>
+                <Typography gutterBottom variant="subtitle1">
+                  {hotel.hotel.name}
+                </Typography>
+                <Typography variant="body2" gutterBottom>
+                  {hotel.hotel.description ? hotel.hotel.description.text
```
Comment (Pix3lAssassin) [inline client/src/components/Hotels/HotelTile.jsx:49]: Clip description length - P2

File: client/src/components/Hotels/Hotels.jsx [hunk 15]
```diff
@@ -1,30 +1,31 @@
...
+  const [hotelInfo, setHotelInfo] = useState({
+    cityCode: '',
+    checkInDate: '',
+    checkOutDate: '',
+    adults: null,
+
+  });
   const classes = useStyles();
+  useEffect(() => {
+  }, [hotels]);
```
Comment (Pix3lAssassin) [inline client/src/components/Hotels/Hotels.jsx:19]: empty useEffect

File: client/src/components/Navigation/Navigation.jsx [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+          {userLogin}
+        </Button>
+        <Link to="/profile">
+          <Button
+            className={classes.loginBtn}
+            style={{ display: profileBtnDisplay }}
+          >
+            {userLogin}
+          </Button>
+        </Link>
```
Comment (Pix3lAssassin) [inline client/src/components/Navigation/Navigation.jsx:62]: should probably use conditional rendering here

File: client/src/components/UserProfile/BookingModals/BookFlight.jsx [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+            label="Email Address"
+            type="email"
+            fullWidth
+            placeholder="Enter Your Email Address"
+            value={email}
+            onChange={(event) => setEmail(event.target.value)}
+            required
+          />
+
+        </DialogContent>
```
Comment (Pix3lAssassin) [inline client/src/components/UserProfile/BookingModals/BookFlight.jsx:113]: Should extract to a new file

File: client/src/components/UserProfile/Budget.jsx [hunk 18]
```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Grid,
+} from '@material-ui/core';
+
+import userProStyles from './UserProStyles.jsx';
```
Comment (Pix3lAssassin) [inline client/src/components/UserProfile/Budget.jsx:7]: typcially imported as useStyles

File: client/src/components/UserProfile/MyTrip.jsx [hunk 19]
```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  Grid,
+} from '@material-ui/core';
+
+import userProStyles from './UserProStyles.jsx';
```
Comment (Pix3lAssassin) [inline client/src/components/UserProfile/MyTrip.jsx:8]: typically imported as useStyles

File: client/src/index.jsx [hunk 20]
```diff
@@ -2,51 +2,87 @@
...
+    width: 'auto',
   },
 }));
 
 function App() {
-  const [cards, setCards] = useState([]);
+  const [user, setUser] = useState('');
+  const [userLogin, setUserLogin] = useState('LOG IN');
+  const [loginBtnDisplay, setLoginBtnDisplay] = useState('true');
+  const [profileBtnDisplay, setProfileBtnDisplay] = useState('true');
```
Comment (Pix3lAssassin) [inline client/src/index.jsx:32]: Refer to last comment on this

**Review by Pix3lAssassin — APPROVED** [review]

File: client/src/components/Flights/FlightController.jsx (hunk 2 above)
Comment (calebbratton) [reply client/src/components/Flights/FlightController.jsx:83]: fixed it (-:

**Pull request merged**